x-pack/plugins/serverless_search/public/application/components/overview_panels/select_client.tsx

```diff
@@ -0,0 +1,181 @@
+            <EuiFlexItem>
+              <EuiImage
+                alt=""
+                src={http.basePath.prepend(`/plugins/${PLUGIN_ID}/assets/${language.iconType}`)}
+                height="100px"
+                width="128px"
+              />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText
+                textAlign="center"
+                color={selectedLanguage === language ? 'default' : 'subdued'}
+              >
+                <h5>
+                  <FormattedMessage
+                    id="xpack.serverlessSearch.selectClient.language.name"
+                    defaultMessage="{languageName}"
+                    values={{
+                      languageName: language.name === Languages.CURL ? 'cURL' : `${language.name}`,
+                    }}
+                  />
+                </h5>
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiPanel>
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  ));
+
+  return (
+    <OverviewPanel
+      description={
+        <FormattedMessage
+          id="xpack.serverlessSearch.selectClient.description"
+          defaultMessage="Elastic builds and maintains clients in several popular languages and our community has contributed many more. Select your favorite language client or dive into the {console} to get started."
+          values={{
+            console: (
+              <EuiLink href="../app/dev_tools#/console">
+                {i18n.translate('xpack.serverlessSearch.selectClient.description.console.link', {
+                  defaultMessage: 'Console',
+                })}
+              </EuiLink>
+            ),
+          }}
+        />
+      }
+      leftPanelContent={
+        <>
+          <EuiFlexGroup gutterSize="l" direction="column">
+            <EuiFlexItem>
+              <EuiText size="s">
+                <strong>
+                  {i18n.translate('xpack.serverlessSearch.selectClient.heading', {
+                    defaultMessage: 'Choose one',
+                  })}
+                </strong>
+              </EuiText>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiFlexGrid gutterSize="l" direction="row" columns={3} alignItems="center">
+                <EuiFlexItem>
+                  <EuiFlexGrid direction="column" gutterSize="s">
+                    <EuiFlexItem>{panelItems[0]}</EuiFlexItem>
+                  </EuiFlexGrid>
+                </EuiFlexItem>
+                <EuiFlexItem>
+                  <EuiFlexGrid direction="column" gutterSize="s">
+                    <EuiFlexItem>{panelItems[1]}</EuiFlexItem>
+                  </EuiFlexGrid>
+                </EuiFlexItem>
+                <EuiFlexItem>
+                  <EuiFlexGrid direction="column" gutterSize="s">
+                    <EuiFlexItem>{panelItems[2]}</EuiFlexItem>
+                  </EuiFlexGrid>
+                </EuiFlexItem>
+              </EuiFlexGrid>
```

Comment:
We should parameterize this instead of repeating it.
```suggestion
              <EuiFlexGrid gutterSize="l" direction="row" columns={panelItems.length} alignItems="center">
              {panelItems.map((panelItem, index) => (
                  <EuiFlexItem key={`panelItem.${index}`}>
                    <EuiFlexGrid direction="column" gutterSize="s">
                      <EuiFlexItem>{panelItem}</EuiFlexItem>
                    </EuiFlexGrid>
                  </EuiFlexItem>
              ))}
              </EuiFlexGrid>
```

Reply:
Nice 💯